src/vipersci/carto/heatmap.py

```diff
@@ -214,12 +192,11 @@ def as_ndarray(input: Sequence) -> NDArray:
 
     if transform is None:
         if sample_bounds is None:
-            minx = np.amin(x_coords_np)
-            maxy = np.amax(y_coords_np)
+            sample_bounds = compute_bounds(x_coords_np, y_coords_np)
```

Comment:
It is extremely bad form to redefine an input variable to be an entirely different kind of object.  The *sample_bounds* object as provided to this function is a shapely.Polygon, but as of this line, it no longer is.  So let's change this variable to just *bounds*.